modules/utxo-lib/test/transaction_util.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * @prettier
+ */
+import * as assert from 'assert';
+import { Network } from '../src/networkTypes';
+
+const Transaction = require('../src/transaction');
+
+export function parseTransactionRoundTrip(buf: Buffer, network: Network) {
+  const tx = Transaction.fromBuffer(buf, network);
+  assert.strictEqual(tx.byteLength(), buf.length);
+  assert.strictEqual(tx.toBuffer().toString('hex'), buf.toString('hex'));
+  assert.strictEqual(tx.clone().toBuffer().toString('hex'), buf.toString('hex'));
```

Comment:
What's the point of cloning the tx and checking it again?

Reply:
making sure that `clone()` is implemented correctly